### PR TITLE
LibGfx/WebPWriter+image: Implement basic lossless webp writing 

### PR DESCRIPTION
**Pull request by nico**
This doesn't use any transforms yet (in particular not the predictor
transform), and doesn't do anything else that actually compresses the
data.

It also give all 256 values code length 8 unconditionally. This means
the huffman trees are not data-dependent at all and provide no
compression. It also means we can just write out the image data
unmodified.

So the output is fairly large. But it _is_ a valid lossless webp file.

Possible follow-ups, to improve compression later:
1. Use actual byte distributions to create huffman trees, to get
   huffman compression.
2. If the distribution has just 1 element, write a simple code length
   code (that way, images that have alpha 0xff everywhere need to store
   no data for alpha).
3. Add backref support, to get full deflate(ish) compression of pixels.
4. Add predictor transform.
5. Consider writing different sets of prefix codes for every 16x16 tile
   (by writing a meta prefix code image).

(It might be nice to make the follow-ups optional, so that this can also
be used as a webp example file generator.)

----

My motivation here is twofold:

1. I currently use the PNG writer for everything, but it is so very slow. Writing a 1024 x 768 bmp (Base/res/wallpapers/sunset-retro.png  converted to bmp) as png takes 1.3s on my system, compared to 0.03s for webp with this writer. Admittedly, the webp is currently uncompressed and 3MB, while the PNG is 1MB. I'm hoping I can learn something about writing encoders and use that to make the PNG writer faster too.

2. I think it'd be nice if we could write lossless animated webp files, and this is a first step towards that.

### CODE REVIEW

File: Userland/Utilities/image.cpp [hunk 1]
```diff
@@ -176,6 +177,10 @@ static ErrorOr<void> save_image(LoadedImage& image, StringView out_path, bool pp
         TRY(Gfx::PortableFormatWriter::encode(*TRY(stream()), *frame, { .format = format }));
         return {};
     }
+    if (out_path.ends_with(".webp"sv, CaseSensitivity::CaseInsensitive)) {
```
Comment (LucasChollet) [inline Userland/Utilities/image.cpp:180]: Guess what? It also needs to be added on line 193 :upside_down_face:

File: Userland/Libraries/LibGfx/ImageFormats/WebPWriter.h [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+#pragma once
+
+#include <AK/Error.h>
+#include <LibGfx/Forward.h>
+
+namespace Gfx {
+
+struct WebPEncoderOptions {
+    Optional<ReadonlyBytes> icc_data;
+};
```
Comment (LucasChollet) [inline Userland/Libraries/LibGfx/ImageFormats/WebPWriter.h:16]: I understand the scaffolding first and meat afterward approach. But I don't know if that's this helpful here, as you do not even implement it in this PR. (It's not a huge deal either, so no change requested)

**Review by LucasChollet — COMMENTED** [review]
Two little comments, otherwise it looks good!

File: Userland/Utilities/image.cpp (hunk 1 above)
Comment (nico) [reply Userland/Utilities/image.cpp:180]: I thought I'd save you the merge conflict… 😅

Done, thanks!

File: Userland/Libraries/LibGfx/ImageFormats/WebPWriter.h (hunk 2 above)
Comment (nico) [reply Userland/Libraries/LibGfx/ImageFormats/WebPWriter.h:16]: I implemented it in this PR at first, but I realized my code was broken and incomplete. The correct implementation is somewhat big (94 insertions(+), 3 deletions(-) -- most of that code is needed for animations too), so I figured I'd put it in a follow-up.

File: Userland/Utilities/image.cpp (hunk 1 above)
Comment (LucasChollet) [reply Userland/Utilities/image.cpp:180]: CLion should be smart enough to figure out how to resolve the conflict

**Pull request merged**